sound/soc/sof/intel/hda-dsp.c

```diff
@@ -696,12 +726,44 @@ int hda_dsp_resume(struct snd_sof_dev *sdev)
 		.state = SOF_DSP_PM_D0,
 		.substate = SOF_HDA_DSP_PM_D0I0,
 	};
-	int ret;
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+	struct hdac_bus *bus = sof_to_bus(sdev);
+	struct hdac_ext_link *hlink = NULL;
+#endif
+	int ret = 0;
 
 	/* resume from D0I3 */
 	if (sdev->dsp_power_state.state == SOF_DSP_PM_D0) {
 		hda_codec_i915_display_power(sdev, true);
 
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+		/* power up links that were active before suspend */
+		list_for_each_entry(hlink, &bus->hlink_list, list) {
+			if (hlink->ref_count) {
+				ret = snd_hdac_ext_bus_link_power_up(hlink);
+				break;
+			}
+		}
+		if (ret < 0) {
+			dev_dbg(sdev->dev,
+				"error %x in %s: failed to power up links",
```

Comment:
nit-pick: we have a convention that dev_err() log start with error: and dev_dbg() don't.
pick one of the two solutions, thanks - same comment for all logs.